### ISSUE TITLE
[0.7.0] Fatal error: Uncaught TypeError > Blog

### ISSUE THREAD

**Issue by Hiobi** [wishthis/wishthis#46]
Hi,

I've got an error on 0.7.0,  reproduced on  _release-candidate_ branch (git pull).
(it seems to be solved on 0.7.1)

- PHP 8.1 / Apache2 ([docker image](https://hub.docker.com/repository/docker/hiob/wishthis) built FROM php:8.1-apache)

```PHP
Fatal error: Uncaught TypeError: wishthis\Cache\Blog::get(): Return value must be of type stdClass|array, null returned in /var/www/html/src/classes/cache/blog.php:45 Stack trace: #0 /var/www/html/src/classes/blog.php(20): wishthis\Cache\Blog->get() #1 /var/www/html/src/classes/blog.php(68): wishthis\Blog::get('https://wishthi...') #2 /var/www/html/src/api/blog.php(26): wishthis\Blog::getPosts() #3 {main} thrown in /var/www/html/src/classes/cache/blog.php on line 45
```

**Comment by grandeljay** [comment]
Thanks for reporting this!

I was hoping nobody would notice but it seems I wasn't fast enough with my fix! As you can already tell, I have fixed the issue on the `develop` branch.

Version 0.7.1 shouldn't be too long now!

**Comment by grandeljay** [comment]
I've pushed all my changes to the RC now, can you please confirm the issue is resolved you for?

**Comment by Hiobi** [comment]
It seems to be solved on 0.7.1 RC. :)

The migration process isn't very obvious : wishthis is by default turn on maintenance mode (it's totally normal) when not login.
 When we try to login we get the error below (MySQL migration needed), at first. To get the migrate to 0.7.1 button, we need to go back to the home. Maybe we can have a message about migration like: "Maintenance mode on, if you are an admin please login" and then, have a redirection to an migration page? 

```PHP

Fatal error: Uncaught PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'expires' in 'field list' in /var/www/html/src/classes/database.php:41 Stack trace: #0 /var/www/html/src/classes/database.php(41): PDO->query('INSERT INTO `se...', 2) #1 /var/www/html/src/pages/login.php(72): wishthis\Database->query('INSERT INTO `se...') #2 /var/www/html/index.php(184): require('/var/www/html/s...') #3 {main} thrown in /var/www/html/src/classes/database.php on line 41
```

**Comment by grandeljay** [comment]
> It seems to be solved on 0.7.1 RC. :)

Great!

> Maybe we can have a message about migration like: "Maintenance mode on, if you are an admin please login" 

Agreed, this would add some clarity. I'll add a prompt to login!

> and then, have a redirection to an migration page?

This should already be the case, I'll double check that and the error of course.

**Comment by grandeljay** [comment]
Hmm.. I'm not able to reproduce the error you have provided.

Did it happen while trying to login during maintenance mode on v0.7.0 (upgrading to v0.7.1)? It seems like your RC is not up-to-date to me, can you please check it?

Does the error still occur?

**Comment by Hiobi** [comment]
> Did it happen while trying to login during maintenance mode on v0.7.0 (upgrading to v0.7.1)?

Yes, after than I updated the docker image and launch the container, wishthis was on maintenance mode. I tried to login and then I've got the error. I had the migration message, when I get back to wishthis's home.

> Does the error still occur?

nope ! It was solved just after migration.

Maybe you're able to reproduce issue only in docker environment, when a db migration is needed.
